tests/triggering/suicide-multi/suite.rc

```diff
@@ -11,9 +11,12 @@
             graph = """
                 fin[-P1] => showdown
 
-                showdown:good => good & ! bad & ! ugly
-                showdown:bad => bad & ! good & ! ugly
-                showdown:ugly => ugly & ! good & ! bad
+                showdown:good => good
```

Comment:
OK. The old logic implies something like `showdown:good & showdown:bad => ! ugly`. Need to find out what that will do!

Reply:
(Obviously, the test has been modified to test the `|` logic. I'll add another test to do the `&` logic for outputs from the same task.)

Reply:
The old `tests/triggering/17-suicide-multi.t` still passes so I'm happy.

Reply:
If you re-factor the original graph to make it a bit easier to understand, it would look like this:

```
showdown:good => good
showdown:bad => bad
showdown:ugly => ugly
showdown:good & showdown:bad => ! ugly
showdown:good & showdown:ugly => ! bad
showdown:bad & showdown:ugly => ! good
```

It does not work, because showdown's script only send one output message at a time, so the `&` output logic should never satisfy the suicide triggers.

Reply:
Triangle logic no longer tested, probably not necessary.

Reply:
I have now modified test 20 to do the original graph, but with new scripts to ensure the correct behaviour for message/output. Note:

```
showdown:good => good & ! bad & ! ugly
showdown:bad => bad & ! good & ! ugly
showdown:ugly => ugly & ! good & ! bad

# is the same as:

showdown:good => good
showdown:good => ! bad
showdown:good => ! ugly
showdown:bad => bad
showdown:bad => ! good
showdown:bad => ! ugly
showdown:ugly => ugly
showdown:ugly => ! good
showdown:ugly => ! bad

# Rearrange:

showdown:good => good
showdown:bad => bad
showdown:ugly => ugly

showdown:good => ! ugly
showdown:bad => ! ugly

showdown:good => ! bad
showdown:ugly => ! bad

showdown:bad => ! good
showdown:ugly => ! good

# Grouping RHS, we have:

showdown:good => good
showdown:bad => bad
showdown:ugly => ugly
showdown:good & showdown:bad => ! ugly
showdown:good & showdown:ugly => ! bad
showdown:bad & showdown:ugly => ! good
```